core/services/job/orm.go

```diff
@@ -77,6 +78,35 @@ func NewORM(db *gorm.DB, config *storm.Config, pipelineORM pipeline.ORM, eventBr
 	}
 }
 
+func PreloadAllJobTypes(db *gorm.DB) *gorm.DB {
+	return db.Preload("FluxMonitorSpec").
+		Preload("DirectRequestSpec").
+		Preload("OffchainreportingOracleSpec").
+		Preload("KeeperSpec").
+		Preload("PipelineSpec").
+		Preload("CronSpec").
+		Preload("WebhookSpec").
+		Preload("VRFSpec")
+}
+
+func PopulateExternalInitiator(db *gorm.DB, jb Job) (Job, error) {
```

Comment:
We do this so that when OnJobDeleted is called it has the ExternalInitiator present. Prior to this PR we were not actually cleaning up the webhook_specs when deleting jobs and thats why we were able to look it up **after** the job was "deleted".